### ISSUE TITLE
Application of generic CSP policy failed

### ISSUE THREAD

**Issue by pwurbs** [rancher/rancher#42486]
**Rancher Server Setup**
- Rancher version: 2.7.4
- Installation option (Docker install/Helm Chart): Chart / RKE1

**Describe the bug**
Applying this additional http header including a generic CSP policy to the ingress cluster config:

`Content-Security-Policy: default-src 'none'; script-src 'self'; connect-src 'self'; img-src 'self'; style-src 'self'; frame-ancestors 'self'; form-action 'self';`

As a result, the Rancher UI was not displayed in the Firefox browser. There are some errors in the browser console regarding blocked inline styles and scripts.

**To Reproduce**
Apply the additional CSP header config to the cluster ingress configuration

**Result**
Web page in Firefox remains empty

**Additional context**
I would expect either a good default CSP header configuration in the nginx based ingress controller or at least a documentation how to setup an appropriate CSP policy to improve security while keep functionality of the Rancher UI.
CSP is a standard to increase client security and Rancher should support their application when using Rancher.

**Comment by bpedersen2** [comment]
As rancher ui uses inline-js you would need to add unsafe-inline ( and drop frame-ancestors). 

hint: using a maximally strict policy on a legacy application is not soemthing I would expect to work out of the box.

**Comment by pwurbs** [comment]
Yes, unsafe-inline must be added for styles and scripts. This is not recommended in the community. Using nonce or hash is nearly not possible, because we can't influence the html content of Rancher. Additionally, we had to allow data: schema. This is the resulting unsatisfying CSP:

`Content-Security-Policy: default-src 'self' data: ; script-src 'self' 'unsafe-inline'; connect-src 'self'; img-src 'self' data: ; style-src 'self' 'unsafe-inline'; frame-ancestors 'self'; form-action 'self';`

**Comment by tariksimsek** [comment]
> Yes, unsafe-inline must be added for styles and scripts. This is not recommended in the community. Using nonce or hash is nearly not possible, because we can't influence the html content of Rancher. Additionally, we had to allow data: schema. This is the resulting unsatisfying CSP:
> 
> `Content-Security-Policy: default-src 'self' data: ; script-src 'self' 'unsafe-inline'; connect-src 'self'; img-src 'self' data: ; style-src 'self' 'unsafe-inline'; frame-ancestors 'self'; form-action 'self';`

Hello I tried this one but grafana is not loading after this policy. I guess i should add or remove an option.

**Comment by tariksimsek** [comment]
I added unsafe-eval option , it worked. I was trying options one by one :) I got it from grafana docs

```
my current Rancher csp: more_set_headers "Content-Security-Policy: default-src 'self' data: ; script-src 'self' 'unsafe-eval' 'unsafe-inline'; connect-src 'self'; img-src 'self' data: ; style-src 'self' 'unsafe-inline'; frame-ancestors 'self'; form-action 'self'";

example from grafana csp :       content_security_policy_template = """script-src 'self' 'unsafe-eval' 'unsafe-inline' 'strict-dynamic' $NONCE;object-src 'none';font-src 'self';style-src 'self' 'unsafe-inline' blob:;img-src * data:;base-uri 'self';connect-src 'self' grafana.com ws://$ROOT_PATH wss://$ROOT_PATH;manifest-src 'self';media-src 'none';form-action 'self';"""


```
